modules/host/host.go

```diff
@@ -245,13 +235,15 @@ func newHost(dependencies dependencies, cs modules.ConsensusSet, tpool modules.T
 		persistDir: persistDir,
 	}
 
+	// Call stop in the event of a partial startup.
```

Comment:
interesting. I think in an ideal startup, you wouldn't spawn any goroutines until all potential failures had been accounted for. But that isn't always possible with a complex object.
Maybe you should go a step further and write a `h.tearDown()` helper? This would close the log and db as well, but only if they were non-nil.